plugins/modules/collaborator_information.py

```diff
@@ -0,0 +1,353 @@
+        for collaborator in collaborators:
+            if(collaborator.login in to_remove):
+                r.remove_from_collaborators(collaborator.login)
+
+
+def change_collaborator_permissions(g, repos, to_change):
+    for repo in repos:
+        r = g.get_repo(repo)
+        collaborator_list = []
+        collaborators = r.get_collaborators(affiliation="direct")
+        for collaborator in collaborators:
+            collaborator_list.append(collaborator.login)
+        for p in to_change:
+            if (p in collaborator_list):
+                r.add_to_collaborators(p, permission=to_change[p])
+
+
+def get_collaborators(g, repo_list):
+    output = dict()
+    for repo in repo_list:
+        dict_repo = list()
+        collab_output = dict()
+        collaborators = g.get_repo(
+            repo).get_collaborators(affiliation="direct")
+        for collaborator in collaborators:
+            collab_output['login'] = collaborator.login
+            collab_output['id'] = collaborator.id
+            collab_output['type'] = collaborator.type
+            collab_output['site_admin'] = collaborator.site_admin
+            permissions = {
+                'triage': collaborator.permissions.triage,
+                'push': collaborator.permissions.push,
+                'pull': collaborator.permissions.pull,
+                'admin': collaborator.permissions.admin
+            }
+            collab_output['permissions'] = permissions
+
+            dict_repo.append(collab_output.copy())
+
+        output[repo] = dict_repo.copy()
+
+    return output
+
+
+def run_module():
+    module_args = dict(
+        token=dict(type='str', default='John Doe'),
+        organization_name=dict(type='str', default='default'),
+        enterprise_url=dict(type='str', default=''),
+        repos=dict(type='list', elements='str'),
+        collaborators_to_add=dict(type='dict'),
+        check_collaborator=dict(type='dict'),
+        collaborators_to_change=dict(type='dict'),
+        collaborators_to_remove=dict(type='list', elements='str')
+    )
+
+    module = AnsibleModule(
+        argument_spec=module_args,
+        supports_check_mode=True
+    )
+
+    result = dict(
+        changed=False,
+        fact=''
+    )
+
+    valid_permissions = ["push", "pull", "admin"]
+
+    # token usage retrieved from module's variables from playbook
+    if(module.params['enterprise_url'] == ''):
+        g = Github(module.params['token'])
+    else:
+        g = Github(module.params['token'],
+                   base_url=module.params['enterprise_url'])
+
+    if len(module.params['repos']):
+        for i in range(len(module.params['repos'])):
+            module.params['repos'][i] = module.params['organization_name'] + \
+                "/" + module.params['repos'][i]
+
+    current_collaborators = get_collaborators(g, module.params['repos'])
+
+    if(module.params['collaborators_to_add']):
+        for permission in module.params['collaborators_to_add']:
```

Comment:
Each time through the loop permission actually holds a user name and not a permission value.  Given this 'permission' seem like a bad name for the loop variable.